### ISSUE TITLE
gcdt installer executable

### ISSUE THREAD

**Issue by markfink** [glomex/gcdt#335]
My experience working in glomex is that often devs find it difficult to use virtualenv and pip. Consequently nobody adventures into more advanced usecases as to develop specialized gcdt plugins. One approach to simplify installation could be to compile an executable which covers the basic usecase (http://www.pyinstaller.org/) for glomex. 

@jbrandstetter, @webratz what do you think

**Comment by webratz** [comment]
Maybe we could offer both ways: The current existing one as a felxible way for advanced users, that want to decide which plugins and versions they use.
And the simplified pyinstaller version where they just use a binary.
With that version we should maybe think about how people update gcdt then

**Comment by Dgadavin** [comment]
I think it's a good idea to have some simple and casual installation process for developers who don't want to learn Python basics. Also, I think will be good to install with gcdt some useful plugins such as:
* config-reader-plugin
* lookups-plugin

**Comment by webratz** [comment]
I'd add all plugins that are not mutually exclusive to the binary build.
Maybe it might make sense to have a glomex internal version with our specific plugins and one public version

**Comment by jbrandstetter** [comment]
Not wanting to learn is a no-go anyway :-) That said, are plugins really exclusive or can we design in a way that they add functionality rather than overwriting existing one? I'm wondering how our bundling for Python will behave in such a binary?

**Comment by markfink** [comment]
the plugin mechanism is new since April 2017. to my knowledge we do not have anything team specific yet besides the glomex-config-reader for hocon format. I believe dp is still on version 0.0.59 so if we do a gcdt executable we should include all plugins but glomex-config-reader.

back in November when we designed the plugin mechanism we have been convinced that devs could manage virtualenv and pip since they need tools like pytest, awsmfa, aswcli anyway.

**Comment by jbrandstetter** [comment]
but wouldn't it be possible to use both config-reader plugins side-by-side?

**Comment by markfink** [comment]
@jbrandstetter yes, that would be possible, too. But this could also lead to a way more confusing situation when smb tries to configure a stack using both json and hocon configuration files at the same time. I think this would not work reliably in all situations and I do not see a good way to fence against it.

@Dgadavin this is not how pip i.e. the gcdt plugin mechanism works. Once a plugin is defined as a dependency you can not "override" it with another plugin. With pip additional dependencies are added. The situation you describe really would mean that you have two plugins installed with conflicting functionality. 
The std. installation you describe should be covered by a "std" `requirements_gcdt.txt` file. I hope that your getting started guide helps to better explain that.

Somehow it appears to be difficult to consistently maintain a list of 7 plugins :(

**Comment by jbrandstetter** [comment]
Where do we really have conflicting plugins? I think we're treating the different config-readers as such but it this the right approach? Shouldn't plugins add/enhance functionality rather than mask/change it? I think it's a valid use case to have different formats of config files.

**Comment by markfink** [comment]
yeah, I am totally open to refactor / build a new plugin mechanism for gcdt. But I think we need to move the discussion for the new / better plugin mechanism to a different ticket: https://github.com/glomex/gcdt/issues/338

**Comment by jbrandstetter** [comment]
fine with that. then a question: would we bundle a set of plugins with this binary installer?

**Comment by markfink** [comment]
first I think we need to agree in which direction we want to go to make the **pip, virtualenv, gcdt plugins** hurdle go away for beginners.

If we want to go down the gcdt binary route 
a) see if we can make this work with OSX and Amazonlinux
b) if this works I would include in the gcdt-binary :
```
gcdt
gcdt-config-reader
gcdt-lookups
gcdt-bundler
gcdt-slack-integration
gcdt-datadog-integration
glomex-checks
```
+ maybe the `conf2json` command for hocon

I would exclude the glomex-config-reader for hocon format and the scaffolding generator(s) since these are more advanced usecases and we want to make it more simple.

**Comment by webratz** [comment]
Another possible way could be having a simple installer shell script like docker has (https://get.docker.com/).
This would then check some preconditions and install stuff the way we intend it.
Advantage of that approach would be that people can easily extend stuff and have no hard cut when  switching from the binary to the pip method
